internal/dag/dag.go

```diff
@@ -862,6 +869,9 @@ type ExtensionCluster struct {
 
 	// ConnectTimeout defines how long the proxy should wait when establishing connection to upstream service.
 	ConnectTimeout time.Duration
+
+	// IdleConnectionTimeout defines how long the proxy should keep idle connection to upstream service before disconnecting.
+	IdleConnectionTimeout timeout.Setting
```

Comment:
ExtensionCluster already has a TimeoutPolicy field, could we collapse this and other timeout fields into there? just did a quick look, not sure if it totally makes sense etc.

Reply:
There was small "side-effect" that maybe makes this change bit illogical:  

`TimeoutPolicy` was previously only for `Route` but now after repurposing the same struct for `Cluster` as well, we will have complete set of timeout values included in both `dag.Route` and `dag.Cluster`. To my surprise, two tests in `builder_test.go` started failing:  the tests were about changing response timeout, which previously meant `Route.TimeoutPolicy.ResponseTimeout` was expected to be set. Now after this change, the test must also expect `Cluster.TimeoutPolicy.ResponseTimeout` to be set, even if response timeout makes no sense in context of `dag.Cluster`.

I'm OK with this small illogicality, but I made the change as separate commit so I can easily revert it as well, if wanted!

Reply:
hm yeah maybe not worth the confusion, thanks for looking into that, ok if you would rather keep it as-is and revert

Reply:
Yeah I'm fine backing it out too, but can live with either way.

Reply:
It doesn't seem like any of these are good options, so I guess we're choosing the least worst?

- Keep what we have here (timeoutpolicy the same, ResponseTimeout must be set for tests)
- revert what we have here, what happens?

Is there a better way to do this? Should we have two timeoutpolicy structs that are mostly the same except for this?

Reply:
> Is there a better way to do this? Should we have two timeoutpolicy structs that are mostly the same except for this?

Exactly @youngnick, I got stalled because I started to think that would be the preferred way: specific timeout struct for route, another for cluster. I then got distracted by other work, so sorry for delay. I should try arrange some time soon! :)

Reply:
Awesome, I'll keep un-staling this then :)

Reply:
Splitting `TimeoutPolicy` into `RouteTimeoutPolicy` and `ClusterTimeoutPolicy` worked OK, but I noticed that `dag.ExtensionCluster` behaves bit like a mixture of route and cluster: in addition of creating a cluster it is also used to configure `ExtAuthz_GrpcService` which  has response timeout just like `Route` does. So to follow the DRY principle, I ended up with following:

* `Route` has timeout policy attribute of type `RouteTimeoutPolicy`
* `Cluster` has `ClusterTimeoutPolicy`
* `ExtensionCluster` has both `RouteTimeoutPolicy` and `ClusterTimeoutPolicy`

and now `Route` does not contain timeout parameters that belong to `Cluster` and the other way around - with the negative side effect that `ExtensionCluster` needed both, but what do you think?